pkg/go/graph/weighted_graph_builder_test.go

```diff
@@ -600,3 +600,952 @@ func TestValidConditionalGraphModel(t *testing.T) {
 	require.Equal(t, Infinite, graph.nodes["job#can_view"].weights["user"])
 	require.Equal(t, 1, graph.nodes["job#owner"].weights["job"])
 }
+
+func TestGraphConstructionOrderedExclusion(t *testing.T) {
+	t.Parallel()
+	model := `
+	model
+  		schema 1.1
+		type user
+		type employee
+        type role
+            relations
+                define assignee: [user]
+        type permission
+            relations
+                define assignee: [employee]
+        type job
+            relations
+                define can_read: [user, employee, role#assignee, permission#assignee] but not cannot_read
+                define cannot_read: [user]
+	`
+	authorizationModel := language.MustTransformDSLToProto(model)
+	wgb := NewWeightedAuthorizationModelGraphBuilder()
+	graph, err := wgb.Build(authorizationModel)
+	require.NoError(t, err)
+
+	require.Len(t, graph.nodes, 10)
+	require.Len(t, graph.edges, 5)
+	exclusionNodeID := graph.edges["job#can_read"][0].to.uniqueLabel
+	require.Len(t, graph.edges[exclusionNodeID], 5)
+	cannotreadID := graph.edges[exclusionNodeID][4].to.uniqueLabel
+	require.Equal(t, "job#cannot_read", cannotreadID)
+}
+
+func TestGraphConstructionDirectAssignation(t *testing.T) {
+	t.Parallel()
+	model := `
+	    model
+                    schema 1.1
+                type folder
+                    relations
+                        define viewer: [user]
+                type user
+	`
+	authorizationModel := language.MustTransformDSLToProto(model)
+	wgb := NewWeightedAuthorizationModelGraphBuilder()
+	graph, err := wgb.Build(authorizationModel)
+	require.NoError(t, err)
+
+	require.Len(t, graph.nodes, 3)
+	require.Len(t, graph.edges, 1)
+	require.True(t, graph.nodes["user"].nodeType == SpecificType)
```

Comment:
Comments to apply other similar pattern.  Use `require.Equal(t, y, x)` instead of `require.True(t, x == y)` as it give better error message when not equal.

```suggestion
	require.Equal(t, SpecificType, graph.nodes["user"].nodeType)
```